### ISSUE TITLE
Reexport `hyper::server::Request`

### ISSUE THREAD

**Issue by cybergeek94** [nickel-org/nickel.rs#368]
Currently in [`multipart`](https://github.com/cybergeek94/multipart) it is necessary to include the `hyper` crate when building the integration with Nickel because it is the `hyper::server::Request` type that is the actual readable part of the request, but this type is not reexported in Nickel.

This causes problems when we want to upgrade the `hyper` version in `multipart`, which we've dealt with before by asking Nickel to upgrade simultaneously. (#367)

However, it would be simpler if Nickel just reexported the `hyper::server::Request` type from the version of Hyper it uses so that it is no longer necessary to import that type in `multipart`'s integration with Nickel. That way, the versions of Hyper and Nickel that Multipart uses don't have to be coupled together.

**Comment by TheNeikos** [comment]
Has there been an update to this?

**Comment by jolhoeft** [comment]
Is this still desired?

**Comment by abonander** [comment]
It would allow me to bring Nickel support back into mainline `multipart`, so yes.

**Comment by jolhoeft** [comment]
Have you had a chance to look at hyper 0.11.x? The Request type changes a fair amount; you can see my work in progress at #410. Would this change still be helpful there?

I'm mostly asking for long term planning. I'll try to fold this into the 0.10.x versions in any case.

**Comment by abonander** [comment]
I'm actually creating a new multipart crate for async HTTP APIs instead of a version bump so the two can develop independently, however, if you're still going to use the Hyper `Request` type in the public API and require going through it to a) get the `Content-Type` header and b) read the request body, then please reexport the `Request` type somewhere so I can get it through Nickel and keep a separate Hyper version.

**Comment by jolhoeft** [comment]
Would the `pub extern crate hyper` be a good solution for both the 0.10.x and 0.11.x development? Since I find I need to access hyper objects in regular use of Nickel, this seems like an all around win. Would this mean that applications that use nickel would no longer need an explicit dependency on hyper too?

**Comment by abonander** [comment]
Yeah, that'd be pretty much perfect. Yes, it wouldn't need a separate Hyper dependency with the correct version pinned.

**Comment by jolhoeft** [comment]
I'll merge #418 tomorrow, and push 0.10.1 to crates.

**Comment by jolhoeft** [comment]
I've pushed 0.10.1 with this change (and some others).
